### PR TITLE
feat: add form to user settings to apply coupon code

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds form to user settings to apply coupon code to user account
- Adds new `coupon_code` property to `DbUser`
- Typo fixes
- Needs https://github.com/open-sauced/api/pull/338

## Notes

Test codes - account must be a non-upgraded role already.

- sauced1
- sauced2
- sauced3
- sauced4
- sauced5

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/engineering/issues/114

## Mobile & Desktop Screenshots/Recordings

image

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by nickytonline — APPROVED** [review]
Thanks for making the change! Left a non-blocking comment. 🚢

File: components/organisms/UserSettingsPage/coupon-form.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        couponCode: code,
+      }),
+    });
+
+    if (response.ok) {
+      toast({ description: "Coupon applied successfully", variant: "success" });
+      setCode("");
+      refreshUser?.();
+    } else {
+      toast({ description: "Unable to apply the coupon!", variant: "danger" });
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply components/organisms/UserSettingsPage/coupon-form.tsx:39]: Yea, that makes more sense. I'll update it to `Coupon not found` as that's the error returned from the API

**Review by jpmcb — COMMENTED** [review]
Couldn't get the `sauced5` coupon to work. I was hitting 404 on the patch to `auth/profile/coupon`:

```sh
curl 'https://beta.api.opensauced.pizza/v1/auth/profile/coupon' \
--compressed \
-X PATCH \
-H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/117.0' -H 'Accept: application/json' \
# redacted headers ...
--data-raw '{"couponCode":"sauced5"}'
```

But then it worked with `sauced1` 🤔 

Ahhhhhhhh, I see: `sauced5` is misspelled in the database:
```
sacued5
```

😅

**Review by jpmcb — APPROVED** [review]
I see my role for my test account has been updated after applying the coupon 👍🏼 

```
select role from users where login = 'john-test-account'

role
bigint
50
```

**Pull request merged**